src/com/trilead/ssh2/crypto/PEMDecoder.java

```diff
@@ -323,7 +335,7 @@ private static PEMStructure parsePEM(char[] pem, CertificateDecoder certificateD
 		return ps;
 	}
 
-	private static void decryptPEM(PEMStructure ps, byte[] pw) throws IOException
+	private static void decryptPEM(PEMStructure ps, char[] pw) throws IOException
```

Comment:
not sure if it is correct to change from bytes to chars

Reply:
See https://docs.oracle.com/javase/8/docs/api/javax/crypto/spec/PBEKeySpec.html for the API I refactored this against.